### PR TITLE
Fix[#2183] inferFlattenedErrors should be a type about input type

### PR DESCRIPTION
**Pull request by ytsunekawa**
This PR is to resolve issue #2183, which is to make inferFlattenedErrors be of type input, even for schemas with transforms defined.
I'm new to writing PRs, so I wasn't sure of the etiquette, and I'm not sure of the differences in the README, so I'd be glad to hear your comments.

### CODE REVIEW

**Comment by aaronadamsCA** [comment]
Just adding a comment to note this isn't stale.

File: deno/lib/README.md
Comment (suhaibmujahid) [inline deno/lib/README.md:1]: The changes in the `README.md` file seem unrelated to what this PR is fixing.
